File: src/python/test/AcceptorTestCase.py
import quickfix as fix
import unittest


class TestApplication(fix.Application):
    def onCreate(self, sessionID): return

    def onLogon(self, sessionID): return

    def onLogout(self, sessionID): return

    def toAdmin(self, sessionID, message): return

    def fromAdmin(self, sessionID, message): return

    def toApp(self, sessionID, message): return

    def fromApp(self, message, sessionID): return


class AcceptorTestCase(unittest.TestCase):

    def setUp(self):
        self.settings = fix.SessionSettings()
        self.sendercompid = "SenderCompID"
        dictionary = fix.Dictionary()
        dictionary.setString(fix.CONNECTION_TYPE, "acceptor")
        dictionary.setBool(fix.USE_DATA_DICTIONARY, False)
        dictionary.setInt(fix.SOCKET_ACCEPT_PORT, 10800)
        dictionary.setBool(fix.SOCKET_REUSE_ADDRESS, True)
        dictionary.setString(fix.SENDERCOMPID, self.sendercompid)
        dictionary.setString(fix.START_TIME, "00:00:00")
        dictionary.setString(fix.END_TIME, "00:00:00")
        dictionary.setString(fix.END_TIME, "00:00:00")
        dictionary.setString(fix.FILE_STORE_PATH, "store")
        dictionary.setString(fix.FILE_LOG_PATH, "log")
        self.settings.set(dictionary)
        sessionID = fix.SessionID("FIX.4.4", self.sendercompid, "TargetCompID")
        global_dict = self.settings.get()
        self.settings.set(sessionID, global_dict)

        self.app = TestApplication()
        self.store_factory = fix.FileStoreFactory(self.settings)
        self.log_factory = fix.ScreenLogFactory(self.settings)
        self.acceptor = fix.SocketAcceptor(self.app, self.store_factory, self.settings, logFactory=None)
        self.acceptor.start()

    def test_add_user(self):
        global_dict = self.settings.get()
        session_id = fix.SessionID("FIX.4.4", "SenderCompID", "TargetCompID-Add")
        self.acceptor.createSession(session_id, global_dict)
        self.assertTrue(self.acceptor.has(session_id))


if __name__ == '__main__':
    unittest.main()

File: src/C++/test/AcceptorTestCase.cpp
/* ====================================================================
* The QuickFIX Software License, Version 1.0
*
* Copyright (c) 2001 ThoughtWorks, Inc.  All rights
* reserved.
*
* Redistribution and use in source and binary forms, with or without
* modification, are permitted provided that the following conditions
* are met:
*
* 1. Redistributions of source code must retain the above copyright
*    notice, this list of conditions and the following disclaimer.
*
* 2. Redistributions in binary form must reproduce the above copyright
*    notice, this list of conditions and the following disclaimer in
*    the documentation and/or other materials provided with the
*    distribution.
*
* 3. The end-user documentation included with the redistribution,
*    if any, must include the following acknowledgment:
*       "This product includes software developed by
*        ThoughtWorks, Inc. (http://www.thoughtworks.com/)."
*    Alternately, this acknowledgment may appear in the software itself,
*    if and wherever such third-party acknowledgments normally appear.
*
* 4. The names "QuickFIX" and "ThoughtWorks, Inc." must
*    not be used to endorse or promote products derived from this
*    software without prior written permission. For written
*    permission, please contact quickfix-users@lists.sourceforge.net.
*
* 5. Products derived from this software may not be called "QuickFIX",
*    nor may "QuickFIX" appear in their name, without prior written
*    permission of ThoughtWorks, Inc.
*
* THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESSED OR IMPLIED
* WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
* OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
* DISCLAIMED.  IN NO EVENT SHALL THOUGHTWORKS INC OR
* ITS CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
* SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
* LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
* USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
* SUCH DAMAGE.
* ====================================================================
*/

#ifdef _MSC_VER
#pragma warning( disable : 4503 4355 4786 )
#include "stdafx.h"
#else
#include "config.h"
#endif

#include "AcceptorTestCase.h"
#include <sstream>

namespace FIX
{
bool AcceptorTestCase::parseStream::onSetup( Acceptor*& pObject )
{
  SessionSettings settings;
  std::string input =
    "[DEFAULT]\n"
    "ConnectionType=acceptor\n"
    "SocketAcceptPort=5000\n"
    "StartTime=00:00:00\n"
    "EndTime=00:00:00\n"
    "UseDataDictionary=N\n"
    "[SESSION]\n"
    "BeginString=FIX.4.2\n"
    "SenderCompID=ISLD\n"
    "TargetCompID=TW\n"
    "[SESSION]\n"
    "BeginString=FIX.4.1\n"
    "SenderCompID=ISLD\n"
    "TargetCompID=WT\n";
  std::stringstream stream( input );
  stream >> settings;
  m_pFactory = new MemoryStoreFactory();
  pObject = new TestAcceptor( settings, *m_pFactory );
  return true;
}

void AcceptorTestCase::parseStream::onRun( Acceptor& object )
{
  assert( object.has( SessionID( BeginString( "FIX.4.2" ),
                                 SenderCompID( "ISLD" ),
                                 TargetCompID( "TW" ) ) ) );
  assert( object.has( SessionID( BeginString( "FIX.4.1" ),
                                 SenderCompID( "ISLD" ),
                                 TargetCompID( "WT" ) ) ) );
}
}
